cpp/modmesh/view/wrap_view.cpp

```diff
@@ -487,44 +459,64 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapRCameraController
-                })
-            .def(
-                "upVector",
-                [](wrapped_base_type & self)
-                {
-                    const auto vector = self.upVector();
-                    return py::make_tuple(vector.x(), vector.y(), vector.z());
                 });
+
+#define DECL_QVECTOR3D_PROPERTY(NAME, GETTER, SETTER)          \
+    .def_property(                                             \
+        #NAME,                                                 \
+        [](wrapped_type & self)                                \
+        {                                                      \
+            QVector3D const v = self.GETTER();                 \
+            return py::make_tuple(v.x(), v.y(), v.z());        \
+        },                                                     \
+        [](wrapped_type & self, std::vector<double> const & v) \
+        {                                                      \
+            double const x = v.at(0);                          \
+            double const y = v.at(1);                          \
+            double const z = v.at(2);                          \
+            self.SETTER(QVector3D(x, y, z));                   \
+        })
+
+        (*this)
+            // clang-format off
+                    DECL_QVECTOR3D_PROPERTY(position, position, setPosition)
+                    DECL_QVECTOR3D_PROPERTY(up_vector, upVector, setUpVector)
+                    DECL_QVECTOR3D_PROPERTY(view_center, viewCenter, setViewCenter)
+                    DECL_QVECTOR3D_PROPERTY(default_position, defaultPosition, setDefaultPosition)
+                    DECL_QVECTOR3D_PROPERTY(default_view_center, defaultViewCenter, setDefaultViewCenter)
+                    DECL_QVECTOR3D_PROPERTY(default_up_vector, defaultUpVector, setDefaultUpVector)
+            // clang-format on
+            ;
+#undef DECL_QVECTOR3D_PROPERTY
+
+#define DECL_FLOAT_PROPERTY(NAME, GETTER, SETTER) \
+    .def_property(                                \
+        #NAME,                                    \
+        [](wrapped_type & self)                   \
+        {                                         \
+            return self.GETTER();                 \
+        },                                        \
+        [](wrapped_type & self, float v)          \
+        {                                         \
+            self.SETTER(v);                       \
+        })
+
+        (*this)
+            // clang-format off
+                DECL_FLOAT_PROPERTY(linear_speed, linearSpeed, setLinearSpeed)
+                DECL_FLOAT_PROPERTY(look_speed, lookSpeed, setLookSpeed)
+                DECL_FLOAT_PROPERTY(default_linear_speed, defaultLinearSpeed, setDefaultLinearSpeed)
+                DECL_FLOAT_PROPERTY(default_look_speed, defaultLookSpeed, setDefaultLookSpeed)
+            // clang-format on
+            ;
+#undef DECL_FLOAT_PROPERTY
```

Comment:
Now use macro instead of defining each getter and setter

Reply:
Long live C pre-processor.  It is very handy in C++.